### ISSUE TITLE
Enhancement: Include more verbose build information in --version

### ISSUE THREAD

**Issue by markjoshwel** [markjoshwel/surplus#20]
I've done it for ganim for fun, so why not add some more information for built releases?

```text
$ s+ -v  # built releases
surplus version 2.0.0 (5bc64ed093@main 2023/09/03 15:50:46)
```

```text
$ s+ -v  # local and git+ installs
surplus version 2.0.0-local (latest@future)
```

on normal operation will not show, but will show when Behaviour debug=True and when run with --version

**Comment by markjoshwel** [comment]
To be included for 2.1.0

**Comment by markjoshwel** [comment]
todo: update __init__.py
